app/models/reaction.rb

```diff
@@ -179,4 +187,18 @@ def negative_reaction_from_untrusted_user?
   def notify_slack_channel_about_vomit_reaction
     Slack::Messengers::ReactionVomit.call(reaction: self)
   end
+
+  def positive_reaction_to_comment_on_own_article?
+    BASE_POINTS.fetch(category, 1.0).positive? &&
+      reactable_type == "Comment" &&
+      reactable&.commentable&.user_id == user_id
```

Comment:
Is this class method used against a collection of reactions? If so, can you confirm that this does not result in N+1 queries?

Also, are we certain that all `commentable`s have a `user_id` and they all point to the `commentable`'s creator (as opposed to denoting a different relationship between the `commentable` and the user)?

Reply:
I adjusted this to only be on initial save — which I believe is the ideal logic anyway as we want this to be point-in-time and not run on updates.... But this also further ensures no future bulk update would affect this. Updated the tests to reflect this logic appropriately.

In general I don't think this would anyway. This only runs on save and we don't have any logic that takes bulk actions like this.

Regarding the `commentable` thing — `user_id` either belongs to the proper author _or_ is `nil` since we can comment directly on podcasts as a commentable. As it stands, there is no `user_id` which refers to anything except the proper author in this context.